### PR TITLE
feat: add custom headers support.

### PR DESCRIPTION
**Pull request by lpegoraro**
# 

```yaml
---
receivers:
  kafka:
    brokers:
    - kafka:9092
    topic: otlp_metrics-sink-id-11
    protocol_version: 2.0.0
extensions:
  pprof:
    endpoint: 0.0.0.0:1888
  basicauth/exporter:
    client_auth:
      username: prom-user
      password: dbpass
exporters:
  prometheusremotewrite:
    endpoint: https://acme.com/prom/push
    headers:
      X-Scope-OrgID: TENANT_1
    auth:
      authenticator: basicauth/exporter
service:
  extensions:
  - pprof
  - basicauth/exporter
  pipelines:
    metrics:
      receivers:
      - kafka
      exporters:
      - prometheusremotewrite

```

### CODE REVIEW

File: maestro/config/exporter_builder.go [hunk 1]
```diff
@@ -21,11 +21,25 @@ type PrometheusExporterConfig struct {
 }
 
 func (p *PrometheusExporterConfig) GetExportersFromMetadata(config types.Metadata, authenticationExtensionName string) (Exporters, string) {
-	endpointCfg := config.GetSubMetadata("exporter")["remote_host"].(string)
+	exporterSubMeta := config.GetSubMetadata("exporter")
+	if exporterSubMeta == nil {
+		return Exporters{}, ""
+	}
+	endpointCfg := exporterSubMeta["remote_host"].(string)
+	customHeaders := exporterSubMeta["headers"]
```
Comment (etaques) [inline maestro/config/exporter_builder.go:29]: nil pointer here if doesnt exist headers index in sink config map[string]interface{}

**Review by etaques — CHANGES_REQUESTED** [review]

File: sinks/backend/prometheus/configuration.go [hunk 2]
```diff
@@ -50,6 +54,16 @@ func (p *Backend) ValidateConfiguration(config types.Metadata) error {
 	if err != nil {
 		return errors.New("must send valid URL for Remote Write")
 	}
+	// check for custom http headers
+	customHeaders, customHeadersOk := config[CustomHeadersConfigFeature]
```
[the remark this answers is not in the document]
Comment (lpegoraro) [reply sinks/backend/prometheus/configuration.go:58]: the customHeadersOk is there for that. there's a check after for the ok.

File: maestro/config/exporter_builder.go (hunk 1 above)
Comment (lpegoraro) [reply maestro/config/exporter_builder.go:29]: that's why in the next line there is a check for customHeaders to be nil.

**Review by etaques — APPROVED** [review]
LGTM

**Pull request merged**